app/models/list.rb

```diff
@@ -27,22 +27,26 @@ class List < ApplicationRecord
 
   def self.viewable(user)
     if user
-      public_scope.or(user.lists).order_by_user(user)
+      public_scope.or(user.lists).order_by_entity_count.order_by_user(user)
     else
       public_scope
     end
   end
 
   def self.editable(user)
     if user&.has_ability?(:list)
-      open_scope.or(user.lists).order_by_user(user)
+      open_scope.or(user.lists).order_by_entity_count.order_by_user(user)
     else
       none
     end
   end
 
+  def self.order_by_entity_count
+    left_joins(:list_entities).group(:id).order('COUNT(ls_list_entity.id) DESC')
```

Comment:
Since you added the new field `entity_count` you might as well use that here too: ` List.order(entity_count: :desc) `

Reply:
Good point.